### PR TITLE
Cherry-pick (a77ed76): Fix AutoExpand Issues with Capitalized $Expand System Query Options

### PR DESCRIPTION
**Pull request by WanjohiSammy**
### Description
Cherry-pick https://github.com/OData/AspNetCoreOData/commit/a77ed76dcb76d9a31e493f23b372c93d397dc061

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Test case modified*
- [x] *Build and test with one-click build and test script passed*

### CODE REVIEW

File: test/Microsoft.AspNetCore.OData.E2E.Tests/AutoExpand/AutoExpandDataModel.cs [hunk 1]
```diff
@@ -64,14 +64,14 @@ public class ZipCodeInfo
...
 {
     public int Id { get; set; }
 
-    [AutoExpand]
-    public Order Order { get; set; }
+[AutoExpand]
+public Order Order { get; set; }
 
-    public People Friend { get; set; }
+    public Person Friend { get; set; }
```
Comment (xuzhg) [inline test/Microsoft.AspNetCore.OData.E2E.Tests/AutoExpand/AutoExpandDataModel.cs:74]: align

**Review by xuzhg — APPROVED** [review]

File: test/Microsoft.AspNetCore.OData.E2E.Tests/AutoExpand/AutoExpandDataModel.cs (hunk 1 above)
Comment (WanjohiSammy) [reply test/Microsoft.AspNetCore.OData.E2E.Tests/AutoExpand/AutoExpandDataModel.cs:74]: @xuzhg This is fixed

**Review by habbes — APPROVED** [review]

File: src/Microsoft.AspNetCore.OData/Query/ODataQueryOptions.cs [hunk 2]
```diff
@@ -869,7 +869,7 @@ internal void AddAutoSelectExpandProperties()
 
     private IDictionary<string, string> GetODataQueryParameters()
     {
-        Dictionary<string, string> result = new Dictionary<string, string>();
+        Dictionary<string, string> result = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```
Comment (habbes) [inline src/Microsoft.AspNetCore.OData/Query/ODataQueryOptions.cs:872]: Is case-insensitivity always supported, or does the user need to toggle some option?

Comment (WanjohiSammy) [reply src/Microsoft.AspNetCore.OData/Query/ODataQueryOptions.cs:872]: @habbes `EnableCaseInsensitive` is by default true: https://github.com/OData/AspNetCoreOData/blob/release-8.x/src/Microsoft.AspNetCore.OData/ODataOptions.cs#L310

image

Comment (WanjohiSammy) [reply src/Microsoft.AspNetCore.OData/Query/ODataQueryOptions.cs:872]: `.AddRouteComponents(..)` will call `BuildRouteContainer(..)` to build service provider with IEdmModel and service collections.

[ODataOptions AddRouteComponents(..)](https://github.com/OData/AspNetCoreOData/blob/release-8.x/src/Microsoft.AspNetCore.OData/ODataOptions.cs#L140)

```cs
builder.Services.AddControllers().AddOData(
    opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(5).AddRouteComponents("odata",EdmModelBuilder.GetEdmModel()));
```

**Pull request merged**